pkg/networkpolicy/packet_test.go

```diff
@@ -64,6 +69,136 @@ func TestUdpFragmentIPv6(t *testing.T) {
 	}
 }
 
+func TestTcpIPv4(t *testing.T) {
+	// A normal TCP session with IPv4
+	tests := []struct {
+		name     string
+		input    []byte
+		err      bool
+		expected packet
+	}{
+		{
+			name:  "SYN",
+			input: packetsTcpIPv4[0][14:],
+			expected: packet{
+				family:  v1.IPv4Protocol,
+				proto:   v1.ProtocolTCP,
+				dstIP:   net.ParseIP("192.168.1.1"),
+				dstPort: 5001,
+				srcIP:   net.ParseIP("192.168.1.201"),
+			},
+		},
+		{
+			name:  "SYN, ACK",
+			input: packetsTcpIPv4[1][14:],
+			expected: packet{
+				family:  v1.IPv4Protocol,
+				proto:   v1.ProtocolTCP,
+				dstIP:   net.ParseIP("192.168.1.201"),
+				srcIP:   net.ParseIP("192.168.1.1"),
+				srcPort: 5001,
+			},
+		},
+		{
+			name:  "ACK 1",
+			input: packetsTcpIPv4[2][14:],
+			expected: packet{
+				family:  v1.IPv4Protocol,
+				proto:   v1.ProtocolTCP,
+				dstIP:   net.ParseIP("192.168.1.1"),
+				dstPort: 5001,
+				srcIP:   net.ParseIP("192.168.1.2+1"),
+			},
+		},
+		{
+			name:  "PSH, ACK",
```

Comment:
we should only be receiving packets for `ctstate new` connections...

Reply:
Yeah, but IMHO `parsePacket()` should be more-or-less generic. It should not panic if some TCP packet arrives that is not a SYN for instance. I think "test too much" here should be ok.